go/client/logclient.go

```diff
@@ -85,13 +82,21 @@ type getAcceptedRootsResponse struct {
 	Certificates []string `json:"certificates"`
 }
 
-// getEntryAndProodReponse represents the JSON response to the CT get-entry-and-proof method
+// getEntryAndProodReponse represents the JSON response to the CT get-entry-and-proof method.
 type getEntryAndProofResponse struct {
 	LeafInput string   `json:"leaf_input"` // the entry itself
 	ExtraData string   `json:"extra_data"` // any chain provided when the entry was added to the log
 	AuditPath []string `json:"audit_path"` // the corresponding proof
 }
 
+// GetProofByHashResponse represents the JSON response to the CT get-proof-by-hash method.
+type GetProofByHashResponse struct {
+	LeafIndex    int64    `json:"leaf_index"` // The 0-based index of the end entity corresponding to the "hash" parameter.
+	AuditPath    [][]byte `json:"audit_path"` // An array of base64-encoded Merkle Tree nodes proving the inclusion of the chosen certificate.
+	Success      bool     `json:"success"`
```

Comment:
I suggest getting rid of this and the ErrorMessage because the GetProofByHashResponse should not be created if there was a failure.